[docs][tutorial]: Update getting started tutorial (#18896)
expo/expo: docs/pages/tutorial/initialize-app.md

You'll need the following tools to get started:

- Install [Expo Go](https://expo.dev/client) on your physical device.
- Prepare your computer by [installing the required tools and dependencies](https://docs.expo.dev/get-started/installation/#requirements).

This tutorial also assumes you have basic knowledge of JavaScript and React or React Native. If you have never written React code, go through [the official React tutorial](https://reactjs.org/tutorial/tutorial.html) first. For more resources, see [additional resources](https://docs.expo.dev/next-steps/additional-resources/).

## Step 1: Initialize a new Expo app

We provide two different ways to get started:

- [Initialize with the starter template](#initialize-with-the-starter-template)
- [Manually initialize with `create-expo-app`](#alternate-manually-initialize-with--create-expo-app)

Choose a way to initialize your app and open the project repository in your favorite code editor or IDE after initializing it. Throughout this tutorial, we use VS Code for our examples.

### Initialize with the starter template

In the directory of your choice with your preferred terminal, download the starter project using an Expo template:

> <Terminal cmd={['$ npx create-expo-app -t expo-tutorial-template']} />

Reviewer: Is the starter template referred to here in a separate PR?

Author: Yeah in this PR: https://github.com/expo/expo/pull/18885#issuecomment-1230195289

---
title: Create your first app
---

import { Terminal } from '~/ui/components/Snippet';
import ImageSpotlight from '~/components/plugins/ImageSpotlight';

In this module, let's learn how to create a new Expo project and explore the app.

## Prerequisites

Before you get started, this tutorial assumes that you:

- Have already installed the [Expo Go](https://expo.dev/client) app on your physical device.

> - Have gone through the [requirements under Installation](https://docs.expo.dev/get-started/installation/#requirements) to set up your development environment.

Author: Great point!

This tutorial also assumes you have basic knowledge of JavaScript and React or React Native. If you have never written React code, go through [the official React tutorial](https://reactjs.org/tutorial/tutorial.html) first. For more resources, see [additional resources](https://docs.expo.dev/next-steps/additional-resources/).

## Step 1: Initialize a new Expo app

For this tutorial, we provide two different ways to get started:

> - [Initialize with the starter template](#initialize-with-the-starter-template)

Author: The explanation is provided in the section. Do you think it lacks something or doesn't provide the message?

Initializing with the default template will still require a user to run `npx create-expo-app`. Do you think an explanation on how Expo template works is required here?

- [Manually initialize with `create-expo-app`](#manually-initialize-with-create-expo-app)

Choose a way to initialize your app and open the project repository in your favorite code editor or IDE after initializing it. Throughout this tutorial, we use VS Code for our examples.

### Initialize with the starter template

In the directory of your choice with your preferred terminal, download the starter project using an Expo template:

<Terminal cmd={['$ npx create-expo-app -t expo-tutorial-template']} />

The starter template you are downloading is an Expo project that comes with:

- All dependencies pre-installed that you can use to build and run the app
- All the assets available in the **assets** directory
- Splash screen and app icon already configured

> You can skip installation steps in further modules mentioned in this tutorial using this path.

Author: Yeah, two ways could lead to confusion. I'll remove this line for now.

> You can skip installation steps in further modules mentioned in this tutorial using this path.

Author: ```suggestion
```

> 
> You can skip installation steps in further modules mentioned in this tutorial using this path.

Author: ```suggestion
```

## Step 2: Run the app on mobile

When you have initialized the app, navigate inside the project directory in your terminal window.

After navigating inside the project directory, run the following command to trigger the [development server](/guides/how-expo-works/#expo-development-server) and see the bare-bones app in action:

<Terminal cmd={['$ npx expo start']} />

After starting the development server, you can open the project in the Expo Go app on your device. Learn more on [how to open the app when developing with Expo on a physical device](/get-started/create-a-new-app/#opening-the-app-on-your-phonetablet).

Here is how the app in the current state, running on an iOS simulator and a physical Android device:

<ImageSpotlight alt="App running on an iOS simulator and a physical Android device." src="/static/images/tutorial/app-running-on-mobile-platforms.jpg" style={{maxWidth: 480}} />

The text displayed on the app's screen above can be found in the **App.js** file of the initialized project. It is the entry point of the app, and is executed when the app starts.

## Step 3: Install dependencies for the web

> When you create a new project with `create-expo-app`, it uses the default [blank template](https://github.com/expo/expo/tree/main/templates/expo-template-blank) that comes pre-configured with necessary dependencies to run your app on mobile platforms. To support the web, you need to add a few additional dependencies to the project. In a terminal window, run the following command to install the necessary dependencies to include web support:

Author: Thanks! I'll replace any further occurrences too in the rest of the chapters.

<Terminal cmd={['$ npx expo install react-native-web@~0.18.7 react-dom@18.0.0 @expo/webpack-config@^0.17.0']} />

> If you have initialized your app with the starter template, you can skip this step. These dependencies are already installed in the template.

Before running the above installation command, make sure to terminate `npx expo start` if it is already running the development server.

## Step 4: Run the app on the web

> After installing the dependencies, run the development server again using `npx expo start`. To see the web app in action, press <kbd>W</kbd> in the terminal window. It will open the web app in your default web browser at a specific URL.

Reviewer 2: ```suggestion
After installing the dependencies, run the development server again using `npx expo start`. To see the web app in action, press <kbd>w</kbd> in the terminal window. It will open the web app in your default web browser at a specific URL.
```

<ImageSpotlight alt="App running on an all platforms" src="/static/images/tutorial/app-running-on-all-platforms.jpg" style={{maxWidth: 720}} />

## Up next

> You have now set up an application in the local environment and launched it. The next step is to learn [how to build the first layout of the app](/tutorial/layout).

Reviewer 2: ```suggestion
You have now initialized an Expo project on your computer and you're ready to start developing StickerSmash. The next step is to learn [how to build the first screen of the app](/tutorial/layout).
```
"screen" feels more natural to me here than "layout"

> If you have trouble opening the app on the device of your choice, refer to the [Opening the app on your phone/tablet](/get-started/create-a-new-app/#opening-the-app-on-your-phonetablet) section.

Reviewer 2: ```suggestion
```
The sentence before this is a good segue to the next chapter. We should put any notes about developing up above.
